### PR TITLE
Add doc for TBPTT

### LINKED ISSUE #19985: Add truncated backpropagation through time (TBPTT) example
### 📚 Documentation

It seems like there is not example anymore about how to deal with Truncated Backpropagation Through Time in the latest versions. Am I wrong?

cc @borda

### PR DESCRIPTION
**Pull request by chualanagit**
## What does this PR do?

Fixes #19985

<details>
  <summary><b>Before submitting</b></summary>

- [x] Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--20422.org.readthedocs.build/en/20422/

### CODE REVIEW

File: docs/source-pytorch/common/lightning_module.rst
Comment (lantiga) [inline docs/source-pytorch/common/lightning_module.rst:1055]: This is actually not a LightningModule attribute and we shouldn't treat it as such.

In my opinion this fits more one of the how to guides https://github.com/Lightning-AI/pytorch-lightning/tree/master/docs/source-pytorch/common (they end up in the How to guides sections in the docs) where you could also have more room to expand on what TBPTT is and maybe have a figure explaining the concept.

**Review by lantiga — COMMENTED** [review]
Thank you for the contribution! I'd recommend moving this section to an How to guide, see comment.

File: docs/source-pytorch/common/lightning_module.rst
Comment (chualanagit) [reply docs/source-pytorch/common/lightning_module.rst:1055]: which doc under `source-pytorch/common` do you think this explanation below to if I may ask?

Comment (lantiga) [reply docs/source-pytorch/common/lightning_module.rst:1055]: I would create a new `tbptt.rst` doc and add details there.
It doesn't need to be large, you can model it after https://github.com/Lightning-AI/pytorch-lightning/blob/master/docs/source-pytorch/common/early_stopping.rst

Comment (chualanagit) [reply docs/source-pytorch/common/lightning_module.rst]: ack

File: docs/source-pytorch/common/tbptt.rst
Comment (lantiga) [inline docs/source-pytorch/common/tbptt.rst:12]: Please add an empty line after code-block or the build will fail

File: docs/source-pytorch/common/tbptt.rst
Comment (lantiga) [inline docs/source-pytorch/common/tbptt.rst:34]: Can you initialize `hiddens` to a concrete value here so the example can run?

**Review by lantiga — COMMENTED** [review]
Thank you! Almost there :-)

**Review by lantiga — APPROVED** [review]
Good to go, thank you @chualanagit!

**Comment by lantiga** [comment]
This will need to be added to the main index but I can do it after this is merged

**Comment by chualanagit** [comment]
> This will need to be added to the main index but I can do it after this is merged

thank you!

**Comment by chualanagit** [comment]
@lantiga can you explain the doc-make errors, these seem to be blocking the merge. I can't seem to figure out what's cause them to fail.

File: docs/source-pytorch/common/tbptt.rst
Comment (lantiga) [inline docs/source-pytorch/common/tbptt.rst:5]:
```suggestion
Truncated Backpropagation Through Time (TBPTT) performs backpropogation every k steps of
```

**Comment by lantiga** [comment]
hey @chualanagit really sorry for the wait on this

we're super close, you got two warnings

### First warning 

`/home/runner/work/pytorch-lightning/pytorch-lightning/docs/source-pytorch/common/tbptt.rst:1: WARNING: Title overline too short.`

just change

```
##############
Truncated Backpropagation Through Time (TBPTT)
##############
```

to

```
##############################################
Truncated Backpropagation Through Time (TBPTT)
##############################################
```

### Second warning

`/home/runner/work/pytorch-lightning/pytorch-lightning/docs/source-pytorch/common/tbptt.rst: WARNING: document isn't included in any toctree`

just add

```
.. displayitem::
    :header: Truncated Back-Propagation Through Time
    :description: Efficiently step through time when training recurrent models
    :button_link: ../common/tbtt.html
    :col_css: col-md-4
    :height: 180
```

to https://github.com/Lightning-AI/pytorch-lightning/blob/master/docs/source-pytorch/common/index.rst

**Comment by lantiga** [comment]
@chualanagit if you give me write access to your fork I can make these changes and get them in

**Comment by chualanagit** [comment]
> hey @chualanagit really sorry for the wait on this
> 
> we're super close, you got two warnings
> 
> ### First warning
> `/home/runner/work/pytorch-lightning/pytorch-lightning/docs/source-pytorch/common/tbptt.rst:1: WARNING: Title overline too short.`
> 
> just change
> 
> ```
> ##############
> Truncated Backpropagation Through Time (TBPTT)
> ##############
> ```
> 
> to
> 
> ```
> ##############################################
> Truncated Backpropagation Through Time (TBPTT)
> ##############################################
> ```
> 
> ### Second warning
> `/home/runner/work/pytorch-lightning/pytorch-lightning/docs/source-pytorch/common/tbptt.rst: WARNING: document isn't included in any toctree`
> 
> just add
> 
> ```
> .. displayitem::
>     :header: Truncated Back-Propagation Through Time
>     :description: Efficiently step through time when training recurrent models
>     :button_link: ../common/tbtt.html
>     :col_css: col-md-4
>     :height: 180
> ```
> 
> to https://github.com/Lightning-AI/pytorch-lightning/blob/master/docs/source-pytorch/common/index.rst

I'll make the change now, one second!

**Comment by chualanagit** [comment]
@lantiga fixed and added you as collaborator as well

**Pull request merged**
